com.pilosa.client/src/test/java/com/pilosa/client/ShardColumnsTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2017 Pilosa Corp.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ * contributors may be used to endorse or promote products derived
+ * from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+package com.pilosa.client;
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.pilosa.client.orm.FieldOptions;
+import com.pilosa.client.orm.Index;
+import com.pilosa.client.orm.IndexOptions;
+
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import static com.pilosa.client.Internal.ImportRequest.parseFrom;
+import static org.junit.Assert.assertEquals;
+
+@Category(UnitTest.class)
+public class ShardColumnsTest {
+
+  private static final String INDEX_NAME = "test-index";
+  private static final String FIELD_NAME = "test-field";
+
+  @Test
+  public void testToImportRequest_csvIndexAndFieldKeys() throws Exception {
```

Comment:
Could you remove the underscore in the method name? Also it would be better if the function signature has the more specialized `InvalidProtocolBufferException` exception.

Reply:
Yes. The methods are updated now, and agreed on having it throw `InvalidProtocolBufferException`. That is also updated.